core/src/main/java/org/apache/brooklyn/core/catalog/internal/CatalogBomScanner.java

```diff
@@ -268,7 +268,7 @@ private void addLibraryDetails(Bundle bundle, Map<String, Object> catalog) {
 
         private String readBom(URL bom) {
             try (final InputStream ins = bom.openStream()) {
-                return Streams.readFullyString(ins);
+                return Streams.readFullyStringAndClose(ins);
```

Comment:
The stream is created in a try-with-resources [1], so no need to close. Is your thinking that if we call `readFullyStringAndClose()` everywhere, then it will be easier to make sure we're closing things everywhere? (e.g. we can search for uses of readFullyString, and not have many usages to review).

[1] https://docs.oracle.com/javase/tutorial/essential/exceptions/tryResourceClose.html